### PR TITLE
Merge CCIP and CCIPReader.

### PR DESCRIPTION
**Pull request by winder**
Remove CCIPReader type alias.

### CODE REVIEW

File: commit/merkleroot/observation_test.go [hunk 1]
```diff
@@ -21,12 +21,13 @@ import (
...
 	"github.com/smartcontractkit/chainlink-ccip/mocks/commit/merkleroot"
 	common_mock "github.com/smartcontractkit/chainlink-ccip/mocks/internal_/plugincommon"
-	reader_mock "github.com/smartcontractkit/chainlink-ccip/mocks/internal_/reader"
+	reader_mock "github.com/smartcontractkit/chainlink-ccip/mocks/pkg/reader"
 
 	"github.com/smartcontractkit/chainlink-ccip/plugintypes"
 )
 
 func Test_Observation(t *testing.T) {
+	t.Skip()
```
Comment (makramkd) [inline commit/merkleroot/observation_test.go:30]: Shouldnt skip?

File: commit/plugin_e2e_test.go [hunk 2]
```diff
@@ -27,6 +28,7 @@ import (
 	"github.com/smartcontractkit/chainlink-ccip/internal/mocks"
 	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
 	reader_mock "github.com/smartcontractkit/chainlink-ccip/mocks/internal_/reader"
+	reader_mock2 "github.com/smartcontractkit/chainlink-ccip/mocks/pkg/reader"
```
Comment (makramkd) [inline commit/plugin_e2e_test.go:31]: pkgreadermocks?

File: commit/merkleroot/observation_test.go (hunk 1 above)
Comment (winder) [reply commit/merkleroot/observation_test.go]: Oops, I was testing something unrelated to this PR

File: internal/plugincommon/ccipreader.go [hunk 3]
```diff
@@ -86,12 +86,12 @@ func (b *BackgroundReaderSyncer) Close() error {
 	return nil
 }
 
-// backgroundReaderSync runs a background process that periodically syncs the provider CCIP reader.
+// backgroundReaderSync runs a background process that periodically syncs the provider CCIPReader.
```
Comment (dimkouv) [inline internal/plugincommon/ccipreader.go:89]: old typo `provided`

Comment (dimkouv) [inline internal/plugincommon/ccipreader.go:89]: old typo `provider -> provided`

**Review by dimkouv — APPROVED** [review]

**Pull request merged**